### ISSUE TITLE
Potential speeding up when disabling cudnn.

### ISSUE THREAD

**Issue by thangvubk** [traveller59/spconv#386]
I found potential speeding up when disabling cudnn by adding ``torch.backends.cudnn.enabled = False`` to main function. I don't know why but tested with V100, RTX 8000. I also run benchmark.py and a U-Net for segmentation, and found significant speeding up.

**Comment by FindDefinition** [comment]
I can't get speed up in benchmark.py when I disable cudnn in my environment (3080 Laptop), I will test this in more cloud GPUs.
This shouldn't happen, our library doesn't use cudnn. The only difference between cudnn on/off should be batch norm.

**Comment by thangvubk** [comment]
Thank you for your reply. It seems related to BN, see [here](https://github.com/pytorch/pytorch/issues/61587). When upgrading pytorch to 1.10, the problem is resolved.

**Comment by FindDefinition** [comment]
I test benchmark.py in V100 and other GPUs:

V100 (cudnn on):
```
basic[f16|ConvAlgo.Native|forward] 0.01792144775390625
basic[f16|ConvAlgo.Native|backward] 0.03387160301208496
basic[f16|ConvAlgo.MaskImplicitGemm|forward] 0.011405134201049804
basic[f16|ConvAlgo.MaskImplicitGemm|backward] 0.01221294403076172
basic[f16|ConvAlgo.MaskSplitImplicitGemm|forward] 0.013409996032714843
basic[f16|ConvAlgo.MaskSplitImplicitGemm|backward] 0.012961196899414062
```
V100 (cudnn off):
```
basic[f16|ConvAlgo.Native|forward] 0.01794285774230957
basic[f16|ConvAlgo.Native|backward] 0.032572460174560544
basic[f16|ConvAlgo.MaskImplicitGemm|forward] 0.011333370208740234
basic[f16|ConvAlgo.MaskImplicitGemm|backward] 0.011572933197021485
basic[f16|ConvAlgo.MaskSplitImplicitGemm|forward] 0.01336665153503418
basic[f16|ConvAlgo.MaskSplitImplicitGemm|backward] 0.01177668571472168
```

RTX A6000 (cudnn on):
```
basic[f16|ConvAlgo.Native|forward] 0.018914151191711425
basic[f16|ConvAlgo.Native|backward] 0.026912212371826172
basic[f16|ConvAlgo.MaskImplicitGemm|forward] 0.011739778518676757
basic[f16|ConvAlgo.MaskImplicitGemm|backward] 0.00927891731262207
basic[f16|ConvAlgo.MaskSplitImplicitGemm|forward] 0.013963150978088378
basic[f16|ConvAlgo.MaskSplitImplicitGemm|backward] 0.00934910774230957
```

3080 Laptop (cudnn on):
```
basic[f16|ConvAlgo.Native|forward] 0.012496066093444825
basic[f16|ConvAlgo.Native|backward] 0.02612614631652832
basic[f16|ConvAlgo.MaskImplicitGemm|forward] 0.010397100448608398
basic[f16|ConvAlgo.MaskImplicitGemm|backward] 0.013266849517822265
basic[f16|ConvAlgo.MaskSplitImplicitGemm|forward] 0.011420178413391113
basic[f16|ConvAlgo.MaskSplitImplicitGemm|backward] 0.011423015594482422
```

3080 Laptop (cudnn off):
```
basic[f16|ConvAlgo.Native|forward] 0.012607312202453614
basic[f16|ConvAlgo.Native|backward] 0.029224443435668945
basic[f16|ConvAlgo.MaskImplicitGemm|forward] 0.010020065307617187
basic[f16|ConvAlgo.MaskImplicitGemm|backward] 0.013532447814941406
basic[f16|ConvAlgo.MaskSplitImplicitGemm|forward] 0.01122286319732666
basic[f16|ConvAlgo.MaskSplitImplicitGemm|backward] 0.011477088928222657
```

WARNING: I find a bug in V100, the backward weight kernel use f16 as accumulator and may cause unstable f16 training, will be fixed soon.
